### PR TITLE
Check realtime mode in remote clock interceptor

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes VIPEROMCT-397

### Describe your changes:
Check that the request is using realtime before using the remote-clock start and end timestamps for telemetry requests

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/remoteClock/requestInterceptor.js [hunk 1]
```diff
@@ -31,10 +31,14 @@ function remoteClockRequestInterceptor(openmct, _remoteClockIdentifier, waitForB
       return activeClock?.key === 'remote-clock' && !remoteClockLoaded;
     },
     invoke: async (request) => {
-      const { start, end } = await waitForBounds();
-      remoteClockLoaded = true;
-      request.start = start;
-      request.end = end;
+      const timeContext = request?.timeContext ?? openmct.time;
```
Comment (ozyx) [inline src/plugins/remoteClock/requestInterceptor.js:34]: ❤️

File: src/plugins/remoteClock/requestInterceptor.js [hunk 2]
```diff
@@ -31,10 +31,14 @@ function remoteClockRequestInterceptor(openmct, _remoteClockIdentifier, waitForB
...
+      //if we're not in realtime mode, we will use the fixed time bounds provided by the request
+      if (timeContext.isRealTime()) {
```
Comment (ozyx) [inline src/plugins/remoteClock/requestInterceptor.js:36]: Should this be `if (!timeContext.isRealTime())` ?

Comment (shefalijoshi) [reply src/plugins/remoteClock/requestInterceptor.js:36]: No. We only want to take the bounds from `waitForBounds()` if we're in realtime mode

Comment (ozyx) [reply src/plugins/remoteClock/requestInterceptor.js:36]: Can we clarify the comment? The way it reads seems confusing.

Comment (jvigliotta) [reply src/plugins/remoteClock/requestInterceptor.js:36]: Yeah, I agree, I think this is good to go after making this comment more clear.

**Review by jvigliotta — CHANGES_REQUESTED** [review]
Just one small change @ozyx pointed out.

**Comment by akhenry** [comment]
VIPEROMCT-397 was closed by this PR - https://github.com/akhenry/openmct-yamcs/issues/369. Do we still need this PR?

**Comment by shefalijoshi** [comment]
> VIPEROMCT-397 was closed by this PR - [akhenry/openmct-yamcs#369](https://github.com/akhenry/openmct-yamcs/issues/369). Do we still need this PR?

Yes we still need this. It was found while I was investigating that bug and you will get incorrect data if you use remote-clock.

**Review by jvigliotta — APPROVED** [review]
LGTM!

**Pull request merged**